lucene/core/src/java/org/apache/lucene/util/bkd/DocIdsWriter.java

```diff
@@ -253,6 +287,33 @@ private static void readDelta16(IndexInput in, int count, int[] docIDs) throws I
     }
   }
 
+  private void readInts21(IndexInput in, int count, int[] docIDs) throws IOException {
+    int i = 0;
+    for (; i < count - 8; i += 9) {
+      long l1 = in.readLong();
+      long l2 = in.readLong();
+      long l3 = in.readLong();
+      docIDs[i] = (int) (l1 >>> 42);
+      docIDs[i + 1] = (int) ((l1 & 0x000003FFFFE00000L) >>> 21);
+      docIDs[i + 2] = (int) (l1 & 0x001FFFFFL);
```

Comment:
Nit: can you remove the unnecessary leading zeros?
```suggestion
      docIDs[i + 2] = (int) (l1 & 0x1FFFFFL);
```